File: apps/api/src/xiaosu/api/health.py
from datetime import UTC, datetime

from fastapi import APIRouter
from pydantic import BaseModel

router = APIRouter(tags=["health"])


class HealthResponse(BaseModel):
    status: str
    service: str
    timestamp: datetime


@router.get("/health", response_model=HealthResponse)
async def health_check() -> HealthResponse:
    return HealthResponse(
        status="ok",
        service="xiaosu-api",
        timestamp=datetime.now(UTC),
    )

File: apps/web/src/api/health.ts
export interface HealthResponse {
  status: string
  service: string
  timestamp: string
}

export async function fetchHealth(): Promise<HealthResponse> {
  const response = await fetch('/api/v1/health')

  if (!response.ok) {
    throw new Error('后端健康检查失败')
  }

  return (await response.json()) as HealthResponse
}

